assets/js/lesson_8/advanced_functions.js

```diff
@@ -0,0 +1,278 @@
+// 1.FIND AVERAGE FUNCTION
+
+let arrayAverage = [2, 3, 4]
+
+// This was my first decision
+/**
+ * @param {number} arr 
+ * @returns {number} 
+ */
+
+function arrayAverageNum(arr) {
+    const onlyNumArray = arr.filter(el => typeof el === "number")
+    const result = onlyNumArray.reduce((a, b) => a + b, 0) / onlyNumArray.length
+
+    return result
+}
+
+console.log(arrayAverageNum(arrayAverage))
+
+/**
+ * But then I thought, what if user will mix string and num input
+ * and numbers will be as string.
+ * And tried this
+ */
+
+//User input
+
+let userInput = true;
+
+while (userInput) {
+    const addElement = prompt('Add element to array. When it is done press "Cancel"')
+
+    if (addElement) {
+        arrayAverage.push(addElement)
+    } else {
+        userInput = confirm('If you added all elements press "Cancel"')
+    }
+}
+
+//Function to calc avg
+
+/**
+ * @param {string} arr 
+ * @returns {number}
+ */
+
+function arrayAverageNum(arr) {
+    const onlyNumArray = arr.filter(Number).map(Number)
+    const result = onlyNumArray.reduce((a, b) => a + b, 0) / onlyNumArray.length
```

Comment:
it could be a separate func to reuse in both functions `arrayAverageNum` and `arrayAverageNum`)))...

PS why do these functions have similar names????? Is it okey??

Reply:
Oh they have similar names because I just copypaste previous function and modified it.  So i forgot to rename it.  I know they are quite similar, but I decided not to delete first one just for the history )